pkg/internal/apis/certmanager/validation/issuer.go

```diff
@@ -335,6 +335,18 @@ func ValidateACMEChallengeSolverDNS01(p *cmacme.ACMEChallengeSolverDNS01, fldPat
 				if len(p.AzureDNS.TenantID) == 0 {
 					el = append(el, field.Required(fldPath.Child("azureDNS", "tenantID"), ""))
 				}
+				if len(p.AzureDNS.ManagedIdentityClientID) > 0 {
+					el = append(el, field.Forbidden(fldPath.Child("azureDNS", "managedIdentityClientID"), "managed identity can not be used at the same time as clientID, tenantID or clientSecret"))
+				}
+				if len(p.AzureDNS.ManagedIdentityResourceID) > 0 {
+					el = append(el, field.Forbidden(fldPath.Child("azureDNS", "managedIdentityResourceID"), "managed identity can not be used at the same time as clientID, tenantID or clientSecret"))
+				}
+			} else {
+				// using managed identity
+				if len(p.AzureDNS.ManagedIdentityClientID) > 0 && len(p.AzureDNS.ManagedIdentityResourceID) > 0 {
+					el = append(el, field.Forbidden(fldPath.Child("azureDNS"), "managedIdentityClientID and managedIdentityResourceID connot both be specified"))
+				}
+
```

Comment:
Please write some unit-tests for these validataion changes. making sure that all the new statements are covered. See https://github.com/jetstack/cert-manager/blob/e343ff5015be5a792bc6df83cef20becffa4aa18/pkg/internal/apis/certmanager/validation/issuer_test.go#L722-L730

And perhaps there should be

Reply:
I've added some tests, let me know if you can think of any other cases I should add